### PR TITLE
Remove redundant information from SdpOrigin. Fixes #14.

### LINKED ISSUE #14: SdpOrigin contains redundant representation of IP address type
[SdpOrigin](https://github.com/nils-ohlmeier/rsdparsa/blob/master/src/lib.rs#L42) has [SdpAddrType](https://github.com/nils-ohlmeier/rsdparsa/blob/master/src/network.rs#L13) which is an enum that tells us if the address is IPv4 or IPv6, but SdpOrigin also has an [IpAddr](https://doc.rust-lang.org/std/net/enum.IpAddr.html) which contains this information as well. This information should be de-duplicated.

### PR DESCRIPTION
**Pull request by Paul-E**

### CODE REVIEW

File: src/network.rs [hunk 1]
```diff
@@ -78,40 +78,26 @@ fn test_parse_addrtype() {
...
+            Err(SdpParserError::Line {
+                message: "Failed to parse unicast address attribute.\
+                          addrtype does not match address."
+                    .to_string(),
+                line: value.to_string()
+            })
+        }
 }
 
 pub fn parse_unicast_addr_unknown_type(value: &str) -> Result<IpAddr, SdpParserError> {
```
Comment (nils-ohlmeier) [inline src/network.rs]: Remove this function all together

File: src/network.rs [hunk 2]
```diff
@@ -78,40 +78,26 @@ fn test_parse_addrtype() {
 }
 
 pub fn parse_unicast_addr(addrtype: &SdpAddrType, value: &str) -> Result<IpAddr, SdpParserError> {
```
Comment (nils-ohlmeier) [inline src/network.rs]: Remove addrtype from the signatur of this function.

Comment (Paul-E) [inline src/network.rs]: There a few places in the code that (seem to) rely on this checking. Should we move the checking to another function or inline the checks? I would be for using a function so there is less duplicate code.

Comment (nils-ohlmeier) [inline src/network.rs]: The addrtype gets parsed before. If the parsing fails the parse_addr_type() function should fail and parse_unicast_addr() should never get called. addrtype is only used in parse_unicast_addr to differentiate which of the two IP address parsers to call. We can get rid of the by simply calling IpAddr::from_str() instead. And then handle the potential error from that in errors.rs.

File: src/network.rs [hunk 3]
```diff
@@ -77,47 +84,18 @@ fn test_parse_addrtype() {
...
+pub fn parse_unicast_addr(value: &str) -> Result<IpAddr, SdpParserError> {
+    IpAddr::from_str(value).map_err(|_| SdpParserError::Line {
+        message: "Failed to parse unicast address attribute"
+            .to_string(),
+        line: value.to_string()
+    })
 }
 
 #[test]
 fn test_parse_unicast_addr_unknown_type() {
```
Comment (nils-ohlmeier) [inline src/network.rs]: Rename these tests so they match the function they test now.

**Pull request merged**